torch/ao/quantization/_dbr/utils.py

```diff
@@ -498,10 +498,11 @@ def get_torch_function_hook_type(
     parent_module: Optional[torch.nn.Module],
     func: Callable,
 ) -> HookType:
-    parent_module_has_qstate = parent_module and \
-        hasattr(parent_module, '_auto_quant_state')
+    parent_module_has_qstate = parent_module is not None and \
+        '_modules' in parent_module.__dict__ and \
+        '_auto_quant_state' in parent_module.__dict__['_modules']
     needs_op_hooks = parent_module_has_qstate and \
-        parent_module._auto_quant_state.cur_op_needs_hooks(func)  # type: ignore[union-attr, operator]
+        parent_module.__dict__['_modules']['_auto_quant_state'].cur_op_needs_hooks(func)  # type: ignore[union-attr, operator]
```

Comment:
wondering should we put write some utility functions ourselves here to make it clearer that we are doing this for performance?

Reply:
added a comment on the reasoning, not sure if adding one more function would increase the overhead, hope that's ok!